frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1479,6 +1490,51 @@ export const dashboardLogic = kea<dashboardLogicType>([
         },
     })),
 
+    subscriptions(({ values, actions }) => ({
+        variables: (variables) => {
+            // try to convert url variables to variables
+            const urlVariables = values.maybeUrlVariables
+            const overrideVariables = []
+
+            for (const [key, value] of Object.entries(urlVariables)) {
+                const variable = variables.find((variable: HogQLVariable) => variable.code_name === key)
+                if (variable) {
+                    overrideVariables.push({ ...variable, value })
+                }
+            }
```

Comment:
So I'm not a fan of allowing users to set any query param and have our logic try to parse it as variables. In the future, we can't use dashboard query params for anything else in case of a conflict of insight variable names. Instead, I think we should have a single query param (something like `query_variables`) that is a key/value pair object instead - this makes it a lot more explicit. 

The downside to this is that it's more effort on the end of building the URL when writing a HogQL query, but we can add the overrides as a parameter of the `<a variableOverrides={...} />` component instead (long term, links are done via chart settings instead of in the query itself)